### PR TITLE
Getting started updates

### PR DESCRIPTION
**Pull request by robbrockbank**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,36 +10,72 @@ We welcome questions/comments/feedback (and pull requests).
...
-## Getting started
+## Getting Started
 
-The recommended way to get started with Calico for the first time is to follow the [Getting Started Guide](docs/GettingStarted.md).  You can be up and running within a matter of minutes.  The guide includes step by step instructions covering
-- setting up a 2 node Linux cluster either
-  - using Vagrant (on Virtual Box or VMware) for easy setup on your laptop or other virtual environment
-  - by manually setting up your own cluster
-- setting up Calico on the cluster
-- creating containers using Calico networking.
+To get started using [Calico Docker networking][calico-networking], we recommend running through one or more of the
```
Comment (tomdee) [inline README.md:15]: Broken link

File: README.md [hunk 2]
```diff
@@ -10,36 +10,72 @@ We welcome questions/comments/feedback (and pull requests).
...
+using Calico. This particular demonstration utilizes the [powerstrip](#docker-with-powerstrip) networking option.
+
+For a quick start, we'd recommend following one of the Vagrant installations using the default docker networking.
+
+
+### Networking options
+
+#### Default docker networking
+
+This uses Dockers standard networking infrastructure, requiring you to explicitly add a created contained into a
```
Comment (tomdee) [inline README.md:45]: typo - contained

File: README.md [hunk 3]
```diff
@@ -10,36 +10,72 @@ We welcome questions/comments/feedback (and pull requests).
...
+Calico network.
```
Comment (tomdee) [inline README.md:46]: I think it's worth mentioning that this is compatible with all docker version from 1.5 onwards.

File: README.md [hunk 4]
```diff
@@ -10,36 +10,72 @@ We welcome questions/comments/feedback (and pull requests).
...
+#### Docker with libnetwork
+
+Docker's native [libnetwork network driver](https://github.com/docker/libnetwork) is available in the Docker
+[experimental channel](https://github.com/docker/docker/tree/master/experimental) alongside the Docker 1.7 release.
+Docker's experimental channel is still moving fast and some of its features are not yet fully stable.  The current
+master branch of calico-docker is largely focussed on interworking with libnetwork.
+
+#### Docker with Powerstrip
+
+Powerstrip is a pluggable HTTP proxy for the Docker API.  Development of the Powerstrip version of Calico is lagging
```
Comment (tomdee) [inline README.md:57]: This shoudl link to the powerstrip repo

File: README.md [hunk 5]
```diff
@@ -10,36 +10,72 @@ We welcome questions/comments/feedback (and pull requests).
...
+#### Default docker networking
+
+This uses Dockers standard networking infrastructure, requiring you to explicitly add a created contained into a
+Calico network.
+
+#### Docker with libnetwork
+
+Docker's native [libnetwork network driver](https://github.com/docker/libnetwork) is available in the Docker
+[experimental channel](https://github.com/docker/docker/tree/master/experimental) alongside the Docker 1.7 release.
+Docker's experimental channel is still moving fast and some of its features are not yet fully stable.  The current
```
Comment (tomdee) [inline README.md:52]: We _could_ mention that setup is a little more involved as it requires the experimental docker version and a separate datastore to cluster docker.

I also don't think it's worth saying that master of calico-docker is focussed on libnetwork - we're planning to pull the libnetwork stuff out into a new repo soon...

File: docs/getting-started/VagrantCoreOS.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+* [VirtualBox][virtualbox] 4.3.10 or greater.
+* [Vagrant][vagrant] 1.6 or greater.
+* [Git][git]
+
+2) Clone this project
+
+    git clone https://github.com/Metaswitch/calico-docker.git
+    
+3) There are three demonstration options depending on whether you are running with libnetwork, powerstrip or the default docker networking.  Select the required demonstration by cd-ing into the appropriate directory:
```
Comment (tomdee) [inline docs/getting-started/VagrantCoreOS.md:18]: Do we need some caps?
And I don't like cd-ing - can you just say changing?

File: docs/getting-started/VagrantCoreOS.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+From calico-02
+
+    ping calico-01
+
+If you see ping failures, the likely culprit is a problem with the VirtualBox network between the VMs.  You should 
+check that each host is connected to the same virtual network adapter in VirtualBox and rebooting the host may also 
+help.  Remember to shut down the VMs with `vagrant halt` before you reboot.
+
+You should also verify that etcd is running and showing both nodes as healthy.
```
Comment (tomdee) [inline docs/getting-started/VagrantCoreOS.md:64]: There is only one node...

File: docs/getting-started/VagrantUbuntu.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
+
+# Set up Calico on CoreOS using Vagrant
```
Comment (tomdee) [inline docs/getting-started/VagrantUbuntu.md:2]: Wrong title

File: docs/getting-started/VagrantUbuntu.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+* [VirtualBox][virtualbox] 4.3.10 or greater.
+* [Vagrant][vagrant] 1.6 or greater.
+* [Git][git]
+
+2) Clone this project
+
+    git clone https://github.com/Metaswitch/calico-docker.git
+    
+3) There are three demonstration options depending on whether you are running with libnetwork, powerstrip or the
+   default docker networking.  Select the required demonstration by cd-ing into the appropriate directory:
```
Comment (tomdee) [inline docs/getting-started/VagrantUbuntu.md:20]: Same comment as coreos

File: docs/getting-started/VagrantUbuntu.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+
+From calico-2
+
+    ping calico-01
+
+If you see ping failures, the likely culprit is a problem with the VirtualBox network between the VMs.  You should 
+check that each host is connected to the same virtual network adapter in VirtualBox and rebooting the host may also 
+help.  Remember to shut down the VMs with `vagrant halt` before you reboot.
+
+You should also verify each host can access etcd.  The following will return an error if etcd is not available.
```
Comment (tomdee) [inline docs/getting-started/VagrantUbuntu.md:65]: The coreos one should have the same thing as here

File: README.md (hunk 1 above)
Comment (tomdee) [inline README.md:15]: I think we should be leading with the demonstrations - the call to action isn't for people to actually run our tutorials, it should be to look at the demonstrations. And then be like "hey! you can run this demo too yourself on all these different platforms"

File: README.md [hunk 11]
```diff
@@ -10,36 +10,72 @@ We welcome questions/comments/feedback (and pull requests).
...
+[networking options](#networking-options), described below.  Each of the demonstrations indicates where the instructions
+differ for each of these networking options.
+
+We also provide a demonstration of a [Docker Swarm cluster](docs/getting-started/powerstrip/CalicoSwarm.md) networked 
+using Calico. This particular demonstration utilizes the [powerstrip](#docker-with-powerstrip) networking option.
+
+For a quick start, we'd recommend following one of the Vagrant installations using the default docker networking.
+
+
+### Networking options
```
Comment (tomdee) [inline README.md:41]: These all need to link through to the demos

**Comment by tomdee** [comment]
I'm happy with the restructure so I'm merging so Tim can rebase. Can you create another PR for the markups?

I've not yet reviewed the AWS/DO/GCE.md files.

File: docs/getting-started/AWS.md [hunk 12]
```diff
@@ -71,42 +103,55 @@ aws ec2 run-instances \
...
+    ssh -i mykey.pem core@<instance IP>
 
 Grab our private IP from the metadata service:
 
-  curl http://169.254.169.254/latest/meta-data/local-ipv4
+    curl http://169.254.169.254/latest/meta-data/local-ipv4
 
 Substitute the private IP obtained above. Run this command on both hosts.
 
-  sudo calicoctl node --ip=<PRIVATE-IP>
```
Comment (tomdee) [inline docs/getting-started/AWS.md:92]: Is Tim fixing this bit?

File: docs/getting-started/DigitalOcean.md [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+
+And repeat this process for a second host, but this time paste in the config from `user-data-other`.
+
+## Set up the IP Pool before running the demo
+SSH into each Calico host you created using the IP addresses found in the Droplets section of the Web Console:
+```
+ssh core@<ip>
+```
+
+On each node, run the calicoctl daemon.
```
Comment (tomdee) [inline docs/getting-started/DigitalOcean.md:47]: Is Tim fixing this bit?

File: docs/getting-started/GCE.md [hunk 14]
```diff
@@ -104,18 +134,33 @@ Outbound rules:
    1 allow
 ```
 
+After creating the WEB profile, run the following command on one of your GCE Calico hosts to create a Calico container
```
Comment (tomdee) [inline docs/getting-started/GCE.md:137]: You need to sort out this paragraph - remove it.

File: docs/getting-started/GCE.md [hunk 15]
```diff
@@ -104,18 +134,33 @@ Outbound rules:
...
+Lastly, the GCE's firewall rules must be updated for any ports you want to expose. Run this gcloud command to allow 
+incoming traffic to port 80:
+
 ```
 gcloud compute firewall-rules create allow-http \
   --description "Incoming http allowed." --allow tcp:80
 ```
 
-You should now be able to access containers using the public ip address of your GCE host on port 80 by visiting http://<host public ip>:80 or running:
+You should now be able to access the NGINX http server using the public ip address of your GCE host on port 80 by
```
Comment (tomdee) [inline docs/getting-started/GCE.md:158]: What NGINX http server?

File: README.md (hunk 3 above)
Comment (robbrockbank) [inline README.md:46]: 1.5 or 1.6?  

Our Environment setup suggests 1.6

**Pull request merged**